matchms/similarity/CosineGreedy.py

```diff
@@ -1,13 +1,63 @@
+from typing import Tuple
 import numpy
 from matchms.typing import SpectrumType
 
 
 class CosineGreedy:
+    """Factory to calculate 'greedy cosine score' between mass spectra.
```

Comment:
```suggestion
    """Factory to calculate 'greedy cosine score' between mass spectra.
    
    The cosine score aims at quantifying the similarity between two mass spectra.
    The score is calculated by finding best possible matches between peaks
    of two spectra. Two peaks are considered a potential match if their
    m/z ratios lie within the given 'tolerance'.
    The underlying peak assignment problem is here solved in a 'greedy' way.
    This can perform notably faster, but does occasionally deviate slightly from
    a fully correct solution (as with the Hungarian algorithm). In practice this
    will rarely affect similarity scores notably, in particular for smaller
    tolerances.
```